internal/prober/http/http.go

```diff
@@ -252,6 +257,28 @@ func convertOAuth2Config(ctx context.Context, cfg *sm.OAuth2Config, logger zerol
 	return r, nil
 }
 
+// Overrides any user-provided headers with our own augmented values
+// for reserved headers.
+func augmentHttpHeaders(check *sm.Check, reservedHeaders http.Header) {
+	headers := []string{}
+	for _, header := range check.Settings.Http.Headers {
+		name, _ := strToHeaderNameValue(header)
+
+		_, present := reservedHeaders[strings.ToLower(name)]
+		if present {
+			continue // users can't override reserved headers with their own values
+		}
+
+		headers = append(headers, header)
+	}
+
+	for header, values := range reservedHeaders {
+		headers = append(headers, fmt.Sprintf("%s:%s", header, strings.Join(values, ",")))
```

Comment:
note that it's OK to have the same header present multiple times

If reservedHeaders has multiple values for a single key, what should be done is add the header multiple times, e.g.

```
for key, values := range reservedHeaders {
   var b strings.Builder
   for _, value := range values {
       b.Reset()
       b.WriteString(key)
       b.WriteRune(':')
       b.WriteString(value)
       headers = append(headers, b.String())
   }
}
```